# Write a python program that asks the user to input a number and prints the reciprocal of that number. 
# Handle the exception if the user input zero
# try:
#     num=int(input("Enter a number:"))
#     num==0
# except:
#     print("zero is not allowed")
# else:    
#     reciprocal=1/num
#     print(reciprocal)



# Write a program that reads a numberfrom the user and prints its square.
# Use the else clause to print a success message
# try:
#     num=int(input("Enter a number:"))
#     result=(num**2)
#     print("The result is:",result)
# except ZeroDivisionError:
#     print("not allowed by negative number")
# else:
#     print("success")




# Write a function that raises a ValueError if the input number is negative
# try:
#     number1=int(input("Enter a number:"))
#     number2=int(input("Enter a number:"))
#     result=number1+number2
#     print(result)
# except ValueError:
#     print("Error occurred please enter a valid number")  
 
 
# MOdify the above program to also handle the exception if the user inputs a non-numeric value 
# try:
#      a=int(input("Enter a number:"))
#      print(a)
# except ValueError:
#     print("This is not a number try again")




# Modify the program in Task 3 to include a finally clause that print a message
# regardless of whether an exception occurred or not
# try:
#     num=int(input("Enter a number:"))
#     result=(num**2)
#     print("The result is:",result)
# except ZeroDivisionError:
#     print("not allowed ny negative number")
# else:
#     print("success")
# finally:
#     print("thank you") 
 

 
# Write a program that repeatedly asks the user for a number and handle exception.
# The program should continue asking untila valid number is entered
# while True:
#     try:
#         num=int(input("Enter a number:"))
#         print(num)
#         break
#     except ValueError:
#         print("invalid number Try again")
    


















    

    








